docs/guides/nrfconnect_factory_data_configuration.md

```diff
@@ -0,0 +1,291 @@
+Factory data is a set of parameters written to flash by a device's manufacturer during the manufacturing process. All of them are written to the non-volatile memory (persistent storage) of microcontroller storage and are protected against rewriting by the software. These parameters should be the same during the whole lifetime of a device, and they can be replaced only by reflashing the entire device using a programmer. 
+
+<p align="center">
+  <img src="../../examples/platform/nrfconnect/doc/images/Logo_RGB_H-small.png" alt="Nordic Semiconductor logo"/>
+  <img src="../../examples/platform/nrfconnect/doc/images/nRF52840-DK-small.png" alt="nRF52840 DK">
+</p>
+
+<hr>
+This document is based on [Matter](https://github.com/project-chip/connectedhomeip), and Nordic
+Semiconductor's nRF Connect SDK, and supports creating factory data.
+
+-   [Overview](#overview)
+    -   [Factory data components](#factory-data-components)
+    -   [Factory data format](#factory-data-format)
+-   [Generating factory data](#generating-factory-data)
+    -   [Creating factory data JSON file](#creating-factory-data-json-file)
+    -   [Verifying using a JSON schema](#verifying-using-a-json-schema)
+    -   [Preparing factory data partition on a device](#preparing-factory-data-partition-on-a-device)
+    -   [Generating factory data partition](#generating-factory-data-partition)
+-   [Building an example with factory data](#building-an-example-with-factory-data)
+-   [Flashing factory data](#flashing-factory-data)
+
+<hr>
+
+<a name="overview"></a>
+
+## Overview
+
+The factory data parameter set includes information such as a manufacturer name, date of manufacturing, product certifications, and PCB versioning. 
+
+On the nRF connect devices, factory data is stored in a Flash memory in a separate partition. This help keep this partition secure by applying non-writable attribute and encryption (Encryption is not implemented yet). Parameters are read at the boot time of a device, and then they can be used 
+in the Matter stack and user application.
+
+### Factory data components
+
+A factory data set consists of following parameters:
+
+|Key name|Full name|Length|Format|Conformance|Description|
+|:------:|:-------:|:----:|:----:|:---------:|:---------:|
+|**`sn`**|`serial number`|<1, 32> B|*ASCII string*|mandatory|A serial number parameter defines an unique number of manufactured device. Maximum length of serial number is 32 characters.|
+|**`vendor_id`**|`vendor ID`|2 B|*uint16*|mandatory|A CHIP-assigned id for the organization responsible for producing the device.|
+|**`product_id`**|`product ID`|2 B|*uint16*|mandatory|A unique id assigned by the device vendor to identify the product or device type. Defaults to a CHIP-assigned id designating a non-production or test "product".|
+|**`vendor_name`**|`vendor name`|<1, 32> B|*ASCII string*|mandatory|A human-readable vendor name which provides a simple string containing identification of device's vendor for the Content APP.|
```

Comment:
I would explain somewhere what is Content APP, or rephrase it, as this term may not be obvious for a user.